src/services/EcoBridge/OmniBridge/OmniBridge.ts

```diff
@@ -0,0 +1,800 @@
+        mode: toTokenMode,
+        mediator: toMediator,
+      },
+    }
+
+    let gas: undefined | string = undefined
+
+    try {
+      const GAS_COST = 260000
+
+      const gasPrice = await this._staticProviders[this._activeChainId]?.getGasPrice()
+
+      if (!gasPrice) throw new Error('Cannot get gas price')
+
+      const {
+        bundle: { nativeCurrencyPrice },
+      } = await request(subgraphClientsUris[this._activeChainId as SWPRSupportedChains], QUERY_ETH_PRICE)
+
+      const gasCostInNativeCurrency = formatEther(gasPrice.mul(GAS_COST))
+
+      gas = `${(Number(gasCostInNativeCurrency) * Number(nativeCurrencyPrice)).toFixed(2)}$`
+    } catch (e) {
+      gas = undefined
+    }
+
+    const details = {
+      fee,
+      gas,
+      receiveAmount: Number(formatUnits(toAmount.toString(), toTokenDecimals)).toFixed(2),
+      estimateTime: '5 min',
+      requestId: helperRequestId,
+    }
+    this.store.dispatch(this.actions.setBridgeDetails(details))
+  }
+
+  private _fetchHistory = async () => {
+    try {
+      if (!this._account || !this._activeChainId) return
+
+      const [{ requests: homeRequests }, { requests: foreignRequests }] = await Promise.all<OmnibridgeSubgraphRequests>(
+        [
+          request(getGraphEndpoint(this._homeChainId, this.bridgeId), requestsUserQuery, {
+            user: this._account,
+          }),
+          request(getGraphEndpoint(this._foreignChainId, this.bridgeId), requestsUserQuery, {
+            user: this._account,
+          }),
+        ]
+      )
+
+      const homeRequestsIds = homeRequests.map(request => request.messageId)
+      const foreignRequestsIds = foreignRequests.map(request => request.messageId)
+
+      const [{ executions: homeExecutions }, { executions: foreignExecutions }] = await Promise.all<
+        OmnibridgeSubgraphExecutions
+      >([
+        request(getGraphEndpoint(this._homeChainId, this.bridgeId), executionsQuery, {
+          messageIds: foreignRequestsIds,
+        }),
+        request(getGraphEndpoint(this._foreignChainId, this.bridgeId), executionsQuery, {
+          messageIds: homeRequestsIds,
+        }),
+      ])
+
+      const homeTransfers = combineTransactions(
+        homeRequests,
+        foreignExecutions,
+        this._homeChainId,
+        this._foreignChainId
+      )
+
+      const foreignTransfers = combineTransactions(
+        foreignRequests,
+        homeExecutions,
+        this._foreignChainId,
+        this._homeChainId
+      )
+
+      this.store.dispatch(this.actions.addTransactions([...homeTransfers, ...foreignTransfers]))
+    } catch (e) {
+      console.warn('Cannot fetch history')
+    }
+  }
+
+  private pendingTxListener = async () => {
+    try {
+      if (!this._account || !this._staticProviders) return
+      const pendingTxs = this.selectors.selectPendingTransactions(this.store.getState(), this._account)
+
+      if (!pendingTxs.length) return
+
+      pendingTxs.map(async pendingTx => {
```

Comment:
I would use for-of loop. Map with async and returning any value seems weird.

Reply:
done [829f8c3](https://github.com/levelkdev/swapr-dapp/pull/1007/commits/829f8c30f9a20ae6b6c797db935bd09f2cad04db)